src/hardware/voodoo.cpp

```diff
@@ -2430,16 +2430,16 @@ do																				\
 	{																			\
 		/* apply dithering */													\
 		APPLY_DITHER(FBZMODE, XX, DITHER_LOOKUP, r, g, b);						\
-		dest[XX] = (uint16_t)((r << 11) | (g << 5) | b);							\
+		(dest)[XX] = (uint16_t)((r << 11) | (g << 5) | b);							\
 	}																			\
 																				\
 	/* write to aux buffer */													\
-	if (depth && FBZMODE_AUX_BUFFER_MASK(FBZMODE))								\
+	if ((depth) && FBZMODE_AUX_BUFFER_MASK(FBZMODE))								\
 	{																			\
 		if (FBZMODE_ENABLE_ALPHA_PLANES(FBZMODE) == 0)							\
-			depth[XX] = (uint16_t)depthval;										\
+			(depth)[XX] = (uint16_t)depthval;										\
 		else																	\
-			depth[XX] = (uint16_t)a;												\
+			(depth)[XX] = (uint16_t)a;												\
```

Comment:
These macros 🥶 Oh man...

Reply:
yup.. the preprocessor expands the macros as one super long line at the point where the macro is called (combined with macros-in-macros further expanding the line). Made it really tricky to understand where static analysis issues were in the code!